azurerm/resource_arm_mariadb_database.go

```diff
@@ -0,0 +1,176 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/mariadb/mgmt/2018-06-01-preview/mariadb"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMariaDbDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMariaDbDatabaseCreateOrUpdate,
+		Read:   resourceArmMariaDbDatabaseRead,
+		Delete: resourceArmMariaDbDatabaseDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[_a-zA-Z0-9]{1,64}$"),
+					"name must be 1 - 64 characters long, and contain only letters, numbers and underscores.",
+				),
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"server_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[-a-zA-Z0-9]{3,50}$"),
+					"server_name must be 3 - 50 characters long, and contain only letters, numbers and hyphens.",
+				),
+			},
+
+			"charset": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[a-z0-9]{3,8}$"),
+					"charset must be 3 - 8 characters long, and contain only lowercase letters and numbers.",
+				),
+			},
+
+			"collation": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCollation(),
+			},
+		},
+	}
+}
+
+func resourceArmMariaDbDatabaseCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbDatabasesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM MariaDB Database creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	serverName := d.Get("server_name").(string)
+
+	charset := d.Get("charset").(string)
+	collation := d.Get("collation").(string)
+
+	properties := mariadb.Database{
+		DatabaseProperties: &mariadb.DatabaseProperties{
+			Charset:   utils.String(charset),
+			Collation: utils.String(collation),
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, serverName, name, properties)
+	if err != nil {
+		return err
```

Comment:
Could we wrap this error?
```suggestion
		return fmt.Errorf("Error creating Maria DB %q (Resource Group %q): %+v", name, resGroup, err)
```